src/main/java/com/minecolonies/entity/ai/citizen/guard/EntityAIMeleeGuard.java

```diff
@@ -0,0 +1,211 @@
+        {
+            return AIState.GUARD_SEARCH_TARGET;
+        }
+        InventoryFunctions.matchFirstInInventory(worker.getInventoryCitizen(), stack -> stack != null && Utils.doesItemServeAsWeapon(stack), worker::setHeldItem);
+        return super.searchTarget();
+    }
+
+    /**
+     * Follow the target and kill it.
+     *
+     * @return the next AIState.
+     */
+    protected AIState huntDown()
+    {
+        if (!targetEntity.isEntityAlive() || checkForWeapon())
+        {
+            targetEntity = null;
+        }
+
+        if (targetEntity != null)
+        {
+            if (worker.getEntitySenses().canSee(targetEntity) && worker.getDistanceToEntity(targetEntity) <= MIN_ATTACK_DISTANCE)
+            {
+                worker.resetActiveHand();
+                attackEntity(targetEntity, (float) DAMAGE_PER_ATTACK);
+                setDelay(getReloadTime());
+                attacksExecuted += 1;
+
+                if (attacksExecuted >= getMaxAttacksUntilRestock())
+                {
+                    return AIState.GUARD_RESTOCK;
+                }
+
+                return AIState.GUARD_HUNT_DOWN_TARGET;
+            }
+            worker.setAIMoveSpeed((float) (BASE_FOLLOW_SPEED + BASE_FOLLOW_SPEED_MULTIPLIER * worker.getExperienceLevel()));
+            worker.isWorkerAtSiteWithMove(targetEntity.getPosition(), (int) MIN_ATTACK_DISTANCE);
+
+            return AIState.GUARD_SEARCH_TARGET;
+        }
+
+        worker.setAIMoveSpeed(1);
+        return AIState.GUARD_SEARCH_TARGET;
+    }
+
+    private void attackEntity(@NotNull EntityLivingBase entityToAttack, float baseDamage)
+    {
+        double damgeToBeDealt = baseDamage;
+
+        ItemStack heldItem = worker.getHeldItem(EnumHand.MAIN_HAND);
+        if (heldItem != null)
+        {
+            if (heldItem.getItem() instanceof ItemSword)
+            {
+                damgeToBeDealt += ((ItemSword) heldItem.getItem()).getDamageVsEntity();
+            }
+            damgeToBeDealt += EnchantmentHelper.getModifierForCreature(heldItem, targetEntity.getCreatureAttribute());
+        }
+
+        targetEntity.attackEntityFrom(new DamageSource(worker.getName()), (float) damgeToBeDealt);
+        targetEntity.setRevengeTarget(worker);
+
+        int fireAspectModifier = EnchantmentHelper.getFireAspectModifier(worker);
+        if (fireAspectModifier > 0)
+        {
+            targetEntity.setFire(fireAspectModifier * FIRE_CHANCE_MULTIPLIER);
+        }
+
+        worker.addExperience(XP_EACH_HIT);
+        worker.faceEntity(entityToAttack, (float) TURN_AROUND, (float) TURN_AROUND);
+        worker.getLookHelper().setLookPositionWithEntity(entityToAttack, (float) TURN_AROUND, (float) TURN_AROUND);
+
+        double xDiff = targetEntity.posX - worker.posX;
+        double zDiff = targetEntity.posZ - worker.posZ;
+
+        double goToX = xDiff > 0 ? MOVE_MINIMAL : -MOVE_MINIMAL;
+        double goToZ = zDiff > 0 ? MOVE_MINIMAL : -MOVE_MINIMAL;
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'goToZ' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)